lib/cocoapods/command/update.rb

```diff
@@ -42,6 +44,14 @@ def initialize(argv)
           end
         end
 
+        unless skipped_pods.empty?
+          if @pods
+            @pods.delete_if { |pod| skipped_pods.include?(pod) }
```

Comment:
I think this should be an error if there are overlapping pod names

Reply:
Thanks for reviewing @segiddins. My understanding was that the `@pods` from lockfile would be the complete list of the installed pods so I need to remove the ones that the user does not want to update. Similar logic would apply in case user specifies the source.
Is the `@pods` variable representing something different? Or maybe some other case I'm not considering?

Reply:
Sorry, I thought it was the list of pods the user had explicitly passed. Eg

`pod update A —skip=A`